airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/context.py

```diff
@@ -318,5 +318,7 @@ def _skip_metadata_disabled_test_suites(self, run_step_options: RunStepOptions)
             RunStepOptions: Updated run step options.
         """
         run_step_options = deepcopy(run_step_options)
-        run_step_options.skip_steps += self._get_step_id_to_skip_according_to_metadata()
+        # If any `skip_steps` are present, we will run everything except the skipped steps, instead of just `keep_steps`.
+        if not run_step_options.keep_steps:
+            run_step_options.skip_steps += self._get_step_id_to_skip_according_to_metadata()
```

Comment:
Thanks for catching this interesting edge case.
We should normally be protected from this issue as the [`__post__init__` of `RunStepOptions`](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pipelines/helpers/execution/run_steps.py#L87) raises an error if we both use `skip_steps` and `keeped_steps`. 
The problem appeared because I my change set `skips_steps` post instantiation.

Your `if` introductions looks good and straightforward to me: if the users passed specific step to only run (via keep steps)  the test skipping logic is bypassed.